### PR TITLE
Auto-detect whether the kernel is bzImage or vmlinuz.

### PR DESCRIPTION
**Pull request by dlorenc**
This changes the extract function to just copy everything it can find, and it moves the detection into the getXhyveArgs function.

What do you think of this approach?

Ref #140

### CODE REVIEW

**Comment by zchee** [comment]
@dlorenc Sorry for delay.
I'll check it later. Thanks!

**Comment by dlorenc** [comment]
> I'll check it later. Thanks!

Thanks, let me know what you think of the approach. Minikube itself won't really know what's in the iso either, so passing this in via a struct parameter would be hard.

File: vmnet/vmnet.go [hunk 1]
```diff
@@ -5,9 +5,10 @@ package vmnet
 import (
 	"fmt"
 	"net"
-	"os"
 	"os/exec"
 	"strings"
+
+	"github.com/docker/machine/libmachine/mcnutils"
```
Comment (zchee) [inline vmnet/vmnet.go]: Why add `"github.com/docker/machine/libmachine/mcnutils"`?
https://github.com/docker/machine/blob/master/libmachine/mcnutils/utils.go is not have IsExist()

Comment (dlorenc) [inline vmnet/vmnet.go]: My mistake, I had to make this public to share it and I accidentally put it in the wrong place...

Comment (zchee) [inline vmnet/vmnet.go]: okay, got it.

**Comment by zchee** [comment]
As in https://github.com/zchee/docker-machine-driver-xhyve/pull/140#issuecomment-257002015
> However, this problem might be solved with adding initialize method.
> Sorry, I still don't understand minikube internal.
> Where do I should add the initialize method for having a backward compatibility of minikube?

I wanted to say *"backward compatibility method"* is this approach. And would be able to provide backward compatibility without changing the minikube.

Hardcoded *"kernel image"* filename is a little bit not good. But It is inevitable.
So almost LGTM.

File: xhyve/xhyve.go [hunk 2]
```diff
@@ -603,22 +603,21 @@ func (d *Driver) extractKernelImages() error {
...
-	}
+	defer func() {
+		log.Debugf("Unmounting %s", isoFilename)
+		err = hdiutil("detach", volumeRootDir)
+	}()
 
-	log.Debugf("Unmounting %s", isoFilename)
-	if err := hdiutil("detach", volumeRootDir); err != nil {
-		return err
+	for _, f := range []string{"vmlinuz64", "bzImage", "initrd", "initrd.img"} {
```
Comment (zchee) [inline xhyve/xhyve.go]: for feature use, I think to add vmlinuz(generally compressed kernel image name) and vmlinux(not compressed) is better.
https://en.wikipedia.org/wiki/Vmlinux
> Traditionally, UNIX platforms called the kernel image /unix. With the development of virtual memory, kernels that supported this feature were given the vm- prefix to differentiate them. The name vmlinux is a mutation of vmunix, while in vmlinuz the letter z at the end denotes that it is compressed (gzipped).[1]

What do you think?

**Comment by zchee** [comment]
@dlorenc Sorry, conflicting `xhyve/xhyve.go`. and WDYT vmlinuz and vmlinux?
Please rebase.

File: xhyve/xhyve.go (hunk 2 above)
Comment (dlorenc) [inline xhyve/xhyve.go]: Not sure I understand. Should I add vmlinux here as well? Or are you suggesting to use that instead of vmlinuz?

Comment (zchee) [inline xhyve/xhyve.go]: Ah, sorry confused.
minikube and boot2docker are used `bzImage` and `vmlinuz64`.
But generally,
- kenrnel filename: `vmlinuz` or `vmlinux` or `bzImage`
- initrd filename: `initrd` or `initrd.gz`

e.g.
http://ftp.debian.org/debian/dists/jessie/main/installer-amd64/current/images/hd-media/

So, I think 
```go
[]string{"vmlinux", "vmlinuz64", "vmlinuz", "bzImage", "initrd", "initrd.gz", "initrd.img"}
```
is better for feature use.

Do you get what I want to say...?

Comment (dlorenc) [inline xhyve/xhyve.go]: Ah, makes sense. Thanks!

**Comment by zchee** [comment]
LTGM. waiting for CI.

**Pull request merged**